transaction/core/src/onetime_keys.rs

```diff
@@ -349,17 +329,6 @@ mod tests {
 
         // Returns meaningless public key.
         assert!(D_prime != *recipient.spend_public_key());
-
-        // view_key_matches_output should return false.
```

Comment:
Any reason why this was removed?

Reply:
its basically a meaningless test, if `D_prime != recipient_spend_public_key`, then view key matches output returns false.

in this change, I'm removing `view_key_matches_output` entirely, and refactoring uses of it in tests to use `D_prime == recipient.spend_public_key`. Since that line is already in the test, there's nothing new to add

Reply:
this is the body of `view_key_matches_output` which is being deleted:

```
pub fn view_key_matches_output(
    view_key: &ViewKey,
    tx_out_target_key: &RistrettoPublic,
    tx_out_public_key: &RistrettoPublic,
) -> bool {
    let D_prime = recover_public_subaddress_spend_key(
        &view_key.view_private_key,
        tx_out_target_key,
        tx_out_public_key,
    );
    view_key.spend_public_key == D_prime
}
```